go.mod

```diff
@@ -7,14 +7,14 @@ require (
 	github.com/gobuffalo/flect v0.2.3
 	github.com/joelanford/go-apidiff v0.1.0
 	github.com/onsi/ginkgo v1.16.5
```

Comment:
Following are some things that we need to do: 

a) We need to clarify that it is breaking change
We also will need to make it very clear when we release the version with the info: https://github.com/kubernetes-sigs/controller-runtime/pull/1773 (because it impacts the scaffolds)

So, I'd like for us to update the title with

⚠️ (go/v3 and config-gen alpha command) Update controller-runtime dep from v0.11.2 to v0.12.0. ( you should explicitly set configmapsleases when configuring leader election, more info: https://github.com/kubernetes-sigs/controller-runtime/pull/1773 )

So that we will have this info for the release notes, it will help to communicate and remember that we need to highlight it to users.

b) Also, we need to update the migration guide from v2 to v3 to have this step so that users that might do the upgrade after we release KB will have the right info: https://book.kubebuilder.io/migration/v2vsv3.html

(it can be a follow-up, could you please track an issue "upgrade controller runtime to the latest ... " so can track there all details and follow-ups?)

c) This PR is missing we upgrade the scaffold of go/v3 plugin: https://github.com/kubernetes-sigs/kubebuilder/blob/master/pkg/plugins/golang/v3/scaffolds/init.go#L33

d) we will also need to run `make generate` and update the downgrade of deps in: https://github.com/kubernetes-sigs/kubebuilder/blob/master/pkg/plugins/golang/v3/commons.go#L89-L151

Reply:
It is solved